lib/oli/authoring/course.ex

```diff
@@ -550,6 +562,46 @@ defmodule Oli.Authoring.Course do
     end
   end
 
+  @doc """
+  Generates a project export for the given project if one is not already in progress
+  """
+  def generate_project_export(project) do
+    case project_export_status(project) do
+      {:in_progress} ->
+        {:error, "Project export is already in progress"}
+
+      _ ->
+        generate_project_export!(project)
+    end
+  end
+
+  defp generate_project_export!(project) do
```

Comment:
This function introduces a bidirectional dependency between this module and the `ProjectExportWorker` module.  Instead, I'd suggest just placing this function directly within the `ProjectExportWorker` module.

Reply:
Good catch, fixed